### PR TITLE
add support for Razer Firefly Hyperflux

### PR DESCRIPTION
**Pull request by hedmo**
my result for adding Razer Firefly Hyperflux

### CODE REVIEW

File: driver/razermousemat_driver.c [hunk 1]
```diff
@@ -627,6 +627,7 @@ static DEVICE_ATTR(matrix_reactive_trigger,  0220, NULL,
 
 
 
+
```
Comment (hedmo) [inline driver/razermousemat_driver.c]: commit 5a27d8b was faulty .removed the line and added it again

**Review by hedmo — COMMENTED** [review]
@z3ntu why wont it pass?

**Comment by z3ntu** [comment]
I've fixed the CI (using `./scripts/format_source.sh`) and squashed your commits into one.

Did you test all effects and do they work properly?

**Comment by hedmo** [comment]
@z3ntu 

i have to be honest it is not working 100% i did not manage to fix matrix and wave effects.I had to set MATRIX_DIMS = [1, 15] as i said in #543.  I have removed the effects  from /daemon/openrazer_daemon/hardware/mouse_mat.py. In RazerGenie it is all good but in polychromatic i can not get custom "effects" to go away (dont think that is a openrazer issue and there are more issues there)

thank you for the help and pulling it and is there any more questions you like to know just type...

best regards hedmo

edit: i had set MATRIX_DIMS to [1, 1] and i have tested them one more time ALL EFFECTS IS WOKING :+1:

**Comment by z3ntu** [comment]
So we can merge this in your opinion?

**Comment by hedmo** [comment]
as you said in  #543:  

> As for effects, Synapse only uses the 'static' effects in hardware it seems, so I can't help you with wave.

go ahead :+1:  i am fine with it and i know people want to have the hyperfux working (even with out wave) . lets make a stamp so i can  terrorize you with new stuff... :)

Edit: in my  opinion... it has been working about a month without any problem and i have  reinstalled openrazer many times after  when i have been working with nari ultimate (spoiler:) and have not seen any bugs or something ells 

regards

File: driver/razermousemat_driver.c [hunk 2]
```diff
@@ -620,11 +650,13 @@ static int razer_mousemat_probe(struct hid_device *hdev, const struct hid_device
     if(intf->cur_altsetting->desc.bInterfaceProtocol == USB_INTERFACE_PROTOCOL_MOUSE) {
 
         switch(usb_dev->descriptor.idProduct) {
+        case USB_DEVICE_ID_RAZER_FIREFLY_HYPERFLUX:
         case USB_DEVICE_ID_RAZER_GOLIATHUS_CHROMA:
         case USB_DEVICE_ID_RAZER_GOLIATHUS_CHROMA_EXTENDED:
             CREATE_DEVICE_FILE(&hdev->dev, &dev_attr_version);
             CREATE_DEVICE_FILE(&hdev->dev, &dev_attr_matrix_custom_frame);
             CREATE_DEVICE_FILE(&hdev->dev, &dev_attr_matrix_reactive_trigger);
+            CREATE_DEVICE_FILE(&hdev->dev, &dev_attr_matrix_effect_wave);
```
Comment (z3ntu) [inline driver/razermousemat_driver.c]: Hmm this adds the wave effect for the Goliathus models as well, but as you said, Wave doesn't work on this so I'll remove it for now

**Comment by z3ntu** [comment]
Thanks a lot for the PR!

**Comment by harry-cpp** [comment]
Hmm, this seems to have only added support for Mamba Hyperflux (the mouse) and not the Firefly Hyperflux (the mouse mat).

(both get identified by the same id, I'm just saying it does not allow me to change the lighting colors from GUI for the mouse mat, maybe I am doing something wrong)

**Comment by harry-cpp** [comment]
I stand corrected to some degree, some leds on the mouse mat do change, to be precise only the top left led, others are always on pink/purple color.

The one led that does change however does not have proper color values applied to it.

**Comment by hedmo** [comment]
@cra0zy 

hi cra0zy 
my mouse mat is lightning up as it should.i hade the same problem first but it  went away.can you change the git to mine?   https://github.com/hedmo/openrazer and i will set MATRIX_DIMS = [1, 15] back to see if this is your problem to 

regards

**Comment by harry-cpp** [comment]
I built it from source and now I see all devices doubled?

**Pull request merged**